x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -179,67 +178,41 @@ export class AlertsClient<
   public async initializeExecution(opts: InitializeExecutionOpts) {
     this.startedAtString = opts.startedAt ? opts.startedAt.toISOString() : null;
 
-    const { runTimestamp, trackedExecutions } = opts;
+    const { runTimestamp } = opts;
 
     if (runTimestamp) {
       this.runTimestampString = runTimestamp.toISOString();
     }
     await this.legacyAlertsClient.initializeExecution(opts);
 
-    this.trackedExecutions = new Set(trackedExecutions ?? []);
-
     // No need to fetch the tracked alerts for the non-lifecycle rules
     if (this.ruleType.autoRecoverAlerts) {
-      const getTrackedAlertsByExecutionUuids = async (executionUuids: string[]) => {
+      const getTrackedAlerts = async () => {
         const result = await this.search({
-          size: (opts.maxAlerts || DEFAULT_MAX_ALERTS) * 2,
-          seq_no_primary_term: true,
+          size: opts.flappingSettings.lookBackWindow,
```

Comment:
Do you think we need the last 20 executions worth of alerts? Alerts from the most recent execution should each carry along their own flapping history and we update "ongoing recovered" alerts for flapping with the latest execution UUID. I'm worried in the worst case, we'll be returning 20 x 1000 alerts whereas previously we'd be returning 2 * 1000. Or am I misunderstanding the query?

Reply:
Yeah I also realized that but lesser size may cause to miss some of the alerts of the last execution.
Yeah in the worst case scenario - if the rule generates 1000 new alerts on each execution- the query returns 20.000 alerts. Under normal circumstances, even if it generates 1000 alerts, they remain as ongoing and only the last execution would carry them.

Actually this is the main difference between this query and the old `getTrackedAlertsByExecutionUuids`. Both returns the alerts of the last 20 executions. But the old one has limit of 2000 for all the alerts.

Reply:
As discussed offline, we should try to find a way to avoid the worst case scenario, where we return 1000 alerts from 20 previous executions.

Possible options:
- splitting the query into 2 - first a collapse query to get the last 20 execution UUIDs (without the inner hits clause) and then a second query to use the execution UUIDs to get alerts (limits the number of alerts that can be returned)
- seeing if there's a way to limit the total size of inner hits returned within the single query
- setting a flag on the ongoing recovered alerts to indicate that they shouldn't be returned for summary alerts queries while still updating the execution UUID to the latest.